site/src/lib/middleware/imageUploadMultipart.ts

```diff
@@ -0,0 +1,40 @@
+import formidable from "formidable";
+import { Middleware } from "next-connect";
+import { PassThrough } from "stream";
+import { BaseApiResponse } from "../handler/baseHandler";
+import { AuthenticatedApiRequest } from "../handler/authenticatedHandler";
+
+export type FormRequestExtensions = {
+  upload?: { file: formidable.File; stream: PassThrough };
+};
+
+export const imageUploadMultipart: Middleware<
+  AuthenticatedApiRequest<FormRequestExtensions>,
+  BaseApiResponse
+> = (req, _res, next) => {
+  const passThrough = new PassThrough();
+
+  const form = new formidable.IncomingForm({
+    fileWriteStreamHandler: () => {
+      return passThrough;
+    },
+    maxFiles: 1,
+    maxFileSize: 5 * 1024 * 1024, // bytes = 5MB
+  });
+
+  const contentType = req.headers["content-type"];
+  if (contentType && contentType.indexOf("multipart/form-data") !== -1) {
+    form.parse(req, (err, _fields, files) => {
+      const file = Object.values(files)[0];
+      // we are only interested with the _first_ file, so other fields are ignored.
```

Comment:
I would return an error somewhere to alert the user to bad input, either:
(a) return an error here if more than one file is provided, or
(b) return an array of files here and let each API route decide how many files it wants (and return an error if there are too many)

Reply:
Not sure what `maxFiles` is doing on the formidable config as adding 2 files to the form doesn't trigger anything